test/test_response.py

```diff
@@ -377,6 +379,74 @@ def test_empty_stream(self):
 
         self.assertRaises(StopIteration, next, stream)
 
+    def test_length_no_header(self):
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, preload_content=False)
+        assert resp.length_remaining is None
+
+    def test_length_w_valid_header(self):
+        headers = {"content-length": "5"}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length_remaining == 5
+
+    def test_length_w_bad_header(self):
+        garbage = {'content-length': 'foo'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length_remaining is None
+
+        garbage['content-length'] = "-10"
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length_remaining is None
+
+    def test_length_when_chunked(self):
+        # This is expressly forbidden in RFC 7230 sec 3.3.2
+        # We fall back to chunked in this case and try to
+        # handle response ignoring content length.
+        headers = {'content-length': '5',
+                   'transfer-encoding': 'chunked'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length_remaining is None
+
+    def test_length_with_multiple_content_lengths(self):
+        headers = {'content-length': '5, 5.0'}
```

Comment:
We shouldn't allow this. Floating point numbers in the content-length field aren't a good idea: we really only want base-10 integers. =)